packages/projen/component/dir-env/src/dirEnv.ts

```diff
@@ -0,0 +1,259 @@
+	DIRENV_VERSION = 'direnv_version',
+	SOURCE_ENV = 'source_env',
+	SOURCE_ENV_IF_EXISTS = 'source_env_if_exists',
+	LOG_STATUS = 'log_status',
+	LOG_ERROR = 'log_error',
+	JOIN_ARGS = 'join_args',
+	EXPAND_PATH = 'expand_path',
+	DOTENV = 'dotenv',
+	DOTENV_IF_EXISTS = 'dotenv_if_exists',
+	ENV_VARS_REQUIRED = 'env_vars_required',
+}
+
+export enum DirEnvLogType {
+	INFO = DirEnvStdLibCommand.LOG_STATUS,
+	ERROR = DirEnvStdLibCommand.LOG_ERROR,
+}
+
+export interface DirEnvOptions {
+	/**
+	 * File name.
+	 */
+	readonly fileName?: string
+}
+
+export class DirEnv extends Component {
+	public static of(project: Project): DirEnv | undefined {
+		const isDirEnv = (c: Component): c is DirEnv => c instanceof DirEnv
+		return project.components.find(isDirEnv)
+	}
+
+	public readonly file: SourceCode
+
+	constructor(project: Project, public readonly options: DirEnvOptions = {}) {
+		super(project)
+		const { fileName = '.envrc' } = options
+		this.#validateFileName(fileName)
+		this.file = new SourceCode(project, fileName, {})
+	}
+
+	#validateFileName(fileName: string) {
+		const regex = new RegExp(/^\.(env|(envrc|envrc\..*))$/)
+		if (!regex.test(fileName)) {
+			throw new Error(`Invalid file name. Does not match ${regex.toString()}`)
+		}
+	}
+
+	#addMarker() {
+		this.addComment(FileBase.PROJEN_MARKER)
+		return this
+	}
+
+	#addLine(line: string) {
+		this.file.line(line)
+	}
+
+	/**
+	 * Open indentation
+	 * @param line
+	 */
+	open(line?: string) {
+		this.file.open(line)
+		return this
+	}
+
+	/**
+	 * Close indentation
+	 * @param line
+	 */
+	close(line?: string) {
+		this.file.close(line)
+		return this
+	}
+
+	/**
+	 * Add an environment variable.
+	 * Optionally add a default value with options.
+	 * Useful for adding sample environment variables.
+	 * @param key
+	 * @param value
+	 * @param options
+	 */
+	addEnvVar(key: string, value: string, options?: { defaultValue: string }) {
+		let line = `export ${key}=${value}`
+		if (typeof options?.defaultValue === 'string') {
+			line = `export ${key}="\${${key}:-${options.defaultValue}}"`
+		}
+		this.#addLine(line)
+		return this
+	}
+
+	/**
+	 * Add a comment.
+	 */
+	addComment(comment: string) {
+		this.#addLine(`# ${comment}`)
+		return this
+	}
+
+	/**
+	 * Add a blank line.
+	 */
+	addBlankLine() {
+		this.#addLine('\n')
+		return this
+	}
+
+	/**
+	 * Add a shebang.
+	 */
+	addSheBang() {
+		this.#addLine('#!/usr/bin/env bash')
+		return this
+	}
+
+	/**
+	 * Add a command.
+	 * @param command
+	 * @param args
+	 */
+	addCommand(command: DirEnvStdLibCommand | string, ...args: string[]) {
+		this.#addLine(`${command} ${args.join(' ')}`)
+		return this
+	}
+
+	/**
+	 * Add status or error log using direnv log_status or log_error helpers.
+	 * @param type
+	 * @param message
+	 */
+	addLog(type: DirEnvLogType, message: string) {
+		this.addCommand(type, `"${message}"`)
+		return this
+	}
+
+	/**
+	 * Add a layout command.
+	 * @param layout
+	 */
+	addLayout(layout: DirEnvLayout) {
```

Comment:
```suggestion
	addLayout(layout: DirEnvLayout | string) {
```
To allow for other layouts missing from enum.